src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,71 @@
+package codeu.controller;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import codeu.model.store.basic.UserStore;
+
+public class AdminServletTest {
+
+  private AdminServlet adminServlet;
+  private HttpServletRequest mockRequest;
+  private HttpSession mockSession;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+
+  @Before
+  public void setup() {
+    adminServlet = new AdminServlet();
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+    
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/admin.jsp"))
+        .thenReturn(mockRequestDispatcher);
+  }
+
+  // Tests accessing the admin page with an unregistered user
+  @Test
+  public void testDoGet_UnregisteredUserReturnsDeniedAccess() throws IOException, ServletException {
+  	Mockito.when(mockSession.getAttribute("user")).thenReturn(null);
```

Comment:
The indentation should be consistent and I believe 2 spaces: https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation